### PR TITLE
PR 47 rebased, updated for MUMPS 5.4.1

### PR DESCRIPTION
**Pull request by dpo**

### CODE REVIEW

**Review by amontoison — APPROVED** [review]
It seems that it still works if we have a Julia structure `mumps_c` that is not exactly the mirror of the C structure.
The attributes at the end of the `mumps_c` structure (`det`, `err`, `_gc_haven` and `_finalized`) are not in the C structure defined in `[sdcz]mumps_c.h`.
The tests passed so let's not change anything.

**Comment by dpo** [comment]
As long as you only add to the end of the struct, it's ok. There are lots of questions in #46 .

File: src/convenience.jl [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+function LinearAlgebra.ldiv!(x,mumps::Mumps,y)
+    suppress_display!(mumps)
+    if mumps.job < 2
+        mumps_factorize!(mumps)
+    end
+    mumps_solve!(x,mumps,y)
+end
+function LinearAlgebra.inv(mumps::Mumps)
+    suppress_display!(mumps)
+    y = sparse(1:mumps.n,1:mumps.n,1:mumps.n,mumps.n,mumps.n)
```
Comment (dpo) [inline src/convenience.jl:385]: This creates a diagonal matrix, but `inv` is supposed to return the entire inverse, isn't it?

**Comment by dpo** [comment]
I fixed a bunch of things, mostly related to return values. Should be getting close now.

**Comment by dpo** [comment]
@amontoison Still approve with the latest changes?

**Comment by amontoison** [comment]
> @amontoison Still approve with the latest changes?

Yes! You can merge the PR.

**Review by amontoison — APPROVED** [review]

**Pull request merged**